src/sentry/snuba/models.py

```diff
@@ -122,6 +122,14 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_querysubscription"
 
+    def remove(self):
+        # TODO(getsentry/team-ospo#190): Prevents a circular import; could probably split up the
+        # source module in such a way that this is no longer an issue.
+        from sentry.snuba.subscriptions import delete_snuba_subscription
+
+        delete_snuba_subscription(self)
+        self.delete()
```

Comment:
You shouldn't delete the subscription here, it happens at the end of the task fired by `delete_snuba_subscription`. Without this row the deletion task will fail.

https://github.com/getsentry/sentry/blob/c10f7b8211690f9e5fbe92ac229f5ea0fc1bc234/src/sentry/snuba/tasks.py#L190

Reply:
🙈 whoops! i only made that as a helper cause i thought the snuba sub didn't delete that afterwards. Cleaned up that code and just used `delete_snuba_subscription` instead (also updated the tests accordingly).